kinode/src/vfs.rs

```diff
@@ -1030,21 +1017,25 @@ fn join_paths_safely(base: &PathBuf, extension: &str) -> PathBuf {
     base.join(extension_path)
 }
 
-async fn handle_fd_request(km: KernelMessage, files: Files) -> anyhow::Result<()> {
+async fn handle_fd_request(km: KernelMessage, files: &mut Files) -> anyhow::Result<()> {
     let Message::Request(Request { body, .. }) = km.message else {
         return Err(anyhow::anyhow!("not a request"));
     };
 
     let request: FdManagerRequest = serde_json::from_slice(&body)?;
 
     match request {
-        FdManagerRequest::Cull {
-            cull_fraction_denominator,
-        } => {
-            let fraction_to_close = files.open_files.len() as u64 / cull_fraction_denominator;
-            files
-                .close_least_recently_used_files(fraction_to_close)
-                .await?;
+        FdManagerRequest::FdsLimit(fds_limit) => {
+            files.fds_limit = fds_limit;
+            if files.open_files.len() as u64 >= fds_limit {
+                crate::fd_manager::send_fd_manager_hit_fds_limit(&files.our, &files.send_to_loop)
+                    .await;
+                files
+                    .close_least_recently_used_files(
+                        (files.open_files.len() as u64 - fds_limit) / 2,
```

Comment:
this isnt quite right: I think it should just be the difference without a denominator